src/pymoca/backends/casadi/model.py

```diff
@@ -332,37 +342,29 @@ def simplify(self, options):
                 old_vars = getattr(self, l)
                 new_vars = []
                 for old_var in old_vars:
-                    if old_var.symbol.numel() > 1:
-                        rows = []
-                        for i in range(old_var.symbol.size1()):
-                            cols = []
-                            for j in range(old_var.symbol.size2()):
-                                if old_var.symbol.size1() > 1 and old_var.symbol.size2() > 1:
-                                    component_symbol = ca.MX.sym('{}[{},{}]'.format(old_var.symbol.name(), i, j))
-                                elif old_var.symbol.size1() > 1:
-                                    component_symbol = ca.MX.sym('{}[{}]'.format(old_var.symbol.name(), i))
-                                elif old_var.symbol.size2() > 1:
-                                    component_symbol = ca.MX.sym('{}[{}]'.format(old_var.symbol.name(), j))
+                    if old_var.symbol._modelica_shape != (1,):
+                        expanded_symbols = []
+                        for ind in np.ndindex(old_var.symbol._modelica_shape):
+                            component_symbol = ca.MX.sym('{}[{}]'.format(old_var.symbol.name(), ",".join(str(i+1) for i in ind)))
+                            component_var = Variable(component_symbol, old_var.python_type)
+                            for attribute in CASADI_ATTRIBUTES:
+                                # Can't convert 3D arrays to MX, so we convert to nparray instead
+                                value = getattr(old_var, attribute)
+                                if not np.isscalar(value):
+                                    value = np.array(value)
```

Comment:
Does this work when value contains symbolic MX?

Reply:
Probably not, but using parameters inside array values is failing for different reasons now anyway.

Reply:
OK, but in any we shouldn't introduce additional array value issues with new code.

Reply:
Possibly, but the only thing I can do now is raise an `NotImplementedError`. I could work the way it is, but with other things (before it would even get here) broken, it is hard to test and be sure.